src/util/tubemap.js

```diff
@@ -676,12 +677,57 @@ function placeReads() {
       if (!element.hasOwnProperty("y")) {
         // previous y value from pathIdx - 1 might not exist yet if that segment is also without node 
         // use previous y value from last segment with node instead 
-        let previousValidY = null;
         let lastIndex = pathIdx - 1;
-        while (previousValidY === null && lastIndex >= 0) {
-          previousValidY = reads[idx].path[lastIndex].y;
+        let previousPathWithNode;
+        while ((previousPathWithNode?.node === null || !previousPathWithNode?.node) && lastIndex >= 0) {
+          previousPathWithNode = reads[idx].path[lastIndex];
           lastIndex = lastIndex - 1;
         }
+
+        let previousValidY = previousPathWithNode?.y;
+
+        // sometimes, elements without nodes are between 2 segments going to a node we've already visited, from the same direction
+        // this means we're looping back to a node we've already been to, and we should sort in reverse
+        
+        // Find the next node in our path
+        let nextPathIndex = pathIdx + 1
+        let nextPathWithNode = reads[idx].path[nextPathIndex];
+        while ((nextPathWithNode?.node === null || !nextPathWithNode?.node) && nextPathIndex < reads[idx].path.length) {
+          nextPathWithNode = reads[idx].path[nextPathIndex];
+          nextPathIndex = nextPathIndex + 1;
+        }
+
+        let nextNodeOrder = nextPathWithNode?.order;
+        let previousNodeOrder = previousPathWithNode?.order;
+
+        // if the previous segment and the next segment is going to the same node
+        let nextNodeJustVisited = previousNodeOrder !== null && nextNodeOrder !== null && previousNodeOrder === nextNodeOrder;
```

Comment:
Are there actual `null`s stored in `order` and/or `isForward` on the visits? Or should these be checking for `undefined` from the optional chaining `?.` instead?